packages/app-bridge/src/events/blockSettings.ts

```diff
@@ -0,0 +1,20 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { BlockSettingsEvents, BlockSettingsUpdateCallback } from '../types';
+
+const APP_BRIDGE_BLOCK_SETTINGS_UPDATED_EVENT = BlockSettingsEvents.BlockSettingsUpdated;
+
+export const onBlockSettingsUpdated = (callback: BlockSettingsUpdateCallback) => {
+    window.emitter.on(APP_BRIDGE_BLOCK_SETTINGS_UPDATED_EVENT, callback);
+};
+
+export const offBlockSettingsUpdated = (callback: BlockSettingsUpdateCallback) => {
+    window.emitter.off(APP_BRIDGE_BLOCK_SETTINGS_UPDATED_EVENT, callback);
+};
```

Comment:
This should be part of the app bridge implementation on clarify. Having this logic inside this package would mean that it's still part of the bundled block.

Reply:
We can have interfaces like this (names can be changed to `{dispatch/handle}Event`:

```
export interface BlockSettingsUpdateEventManager {
    onUpdate: (event: BlockSettingsUpdateEventCallback) => void;
    offUpdate: (event: BlockSettingsUpdateEventCallback) => void;
    notifyUpdate: (event: BlockSettingsUpdateEventCallback) => void;
}
```
and the implementation can be injected into the hook like:
```
export const useBlockSettings = <T = Record<string, unknown>>(
    appBridge: AppBridgeBlock,
    blockSettingsEvent: BlockSettingsUpdateEventManager,
): [T, (newSettings: Partial<T>) => Promise<void>] => {
    ....
    
    useEffect(() => {
        blockSettingsEvent.onUpdate(updateBlockSettingsFromEvent);
        
        return () => {
            blockSettingsEvent.offUpdate(updateBlockSettingsFromEvent);
        };
    
    const updateBlockSettings = async (blockSettingsUpdate: Partial<T>) => {
        try {
            await appBridge.updateBlockSettings(blockSettingsUpdate);
            blockSettingsEvent.notifyUpdate({
                blockId,
                blockSettings: mergeDeep(blockSettings, blockSettingsUpdate),
            });
        } catch (error) {
            console.error(error);
        }
    };
    
    return [blockSettings, updateBlockSettings];
    
}
```

The implementation in `clarify`:
```
import { BlockSettingsUpdateEventManager } from '@frontify/app-bridge';

export class BlockSettingsUpdateEvents implements BlockSettingsUpdateEventManager {
    public onUpdate(callback: (event: BlockSettingsUpdateEvents) => void): void {
        window.emitter.on('AppBridge:BlockSettingsUpdated', callback);
    }

    public offUpdate(callback: (event: BlockSettingsUpdateEvents) => void): void {
        window.emitter.off('AppBridge:BlockSettingsUpdated', callback);
    }

    public notifyUpdate(updatedBlockSettings: Record<string, unknown>): void {
        window.emitter.emit('AppBridge:BlockSettingsUpdated', updatedBlockSettings);
    }
}
```

Reply:
> Since we are going to change the interface with the command pattern anyways, it's probably not necessary to move these emitters until we've finished it.

Sorry, didn't notice this comment. Agreed. The example I shared above can be the _final_ version and iteratively I can start by building interfaces + implementation without touching the hooks (and replace only when finished)